apiserver/charms.go

```diff
@@ -78,7 +78,7 @@ func (h *charmsHandler) serveGet(w http.ResponseWriter, r *http.Request) error {
 	// Retrieve or list charm files.
 	// Requires "url" (charm URL) and an optional "file" (the path to the
 	// charm file) to be included in the query.
-	charmArchivePath, fileArg, err := h.processGet(r, st)
+	charmArchivePath, fileArg, serveDefaultIcon, err := h.processGet(r, st)
```

Comment:
Perhaps "serveIcon" would be a more appropriate name, as that's the name of the
query value and what we're really doing is asking for the icon, which a fallback to
the default icon only when needed.